### ISSUE TITLE
Error /bin/sh: 1: ../env/bin/python: not found when execute bench --site command migrate before starting the custom app

### ISSUE THREAD

**Issue by dungtc** [frappe/frappe_docker#1187]
## Description of the issue
Build custom app and start to run `bench --site my_site migrate` before running the image I got this error

Reason: because the WORKDIR is **"/home/frappe/frappe-bench"** and the bench bin_path was configured with absolute path **"../env/bin"** in https://github.com/frappe/frappe/blob/develop/frappe/commands/utils.py#L565C15-L565C16 

The requirement path must be **"/home/frappe/frappe-bench/sites"**

### Expected result
Change the WORKDIR of custom app from **"/home/frappe/frappe-bench"** to **"/home/frappe/frappe-bench/sites"** to run bench commands before starting the app

### Stacktrace / full error message if available
Run `bench --site my_site migrate` before starting the custom app
```
"/bin/sh: 1: ../env/bin/python: not found"
```

**Comment by revant** [comment]
Please add a failing test.

https://github.com/frappe/frappe_docker/actions/runs/5617087162/job/15220580513

**Comment by dungtc** [comment]
Hi @revant i still got this error when i run worker on erpnext helm chart.
This happened since the new custom app build way was updated.

Stacktrace

```
rv = self.invoke(ctx)
      self = <Group None>
      args = ['frappe', '--site', 'localhost', 'migrate']
      prog_name = 'bench'
      complete_var = None
      standalone_mode = True
      extra = {}
      ctx = <click.core.Context object at 0x7fd222945e50>
  File "/home/frappe/frappe-bench/env/lib/python3.11/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
      self = <Group None>
      ctx = <click.core.Context object at 0x7fd222945e50>
      _process_result = <function MultiCommand.invoke.<locals>._process_result at 0x7fd222961d00>
      args = ['migrate']
      cmd_name = 'frappe'
      cmd = <Group frappe>
      sub_ctx = <click.core.Context object at 0x7fd223ba5f10>
  File "/home/frappe/frappe-bench/env/lib/python3.11/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
      self = <Group frappe>
      ctx = <click.core.Context object at 0x7fd223ba5f10>
      _process_result = <function MultiCommand.invoke.<locals>._process_result at 0x7fd22296e0c0>
      args = []
      cmd_name = 'migrate'
      cmd = <Command migrate>
      sub_ctx = <click.core.Context object at 0x7fd2227dfad0>
  File "/home/frappe/frappe-bench/env/lib/python3.11/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
      self = <Command migrate>
      ctx = <click.core.Context object at 0x7fd2227dfad0>
  File "/home/frappe/frappe-bench/env/lib/python3.11/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
      args = ()
      kwargs = {'skip_failing': False, 'skip_search_index': False}
      self = <click.core.Context object at 0x7fd2227dfad0>
      callback = <function migrate at 0x7fd2227f4900>
      ctx = <click.core.Context object at 0x7fd2227dfad0>
  File "/home/frappe/frappe-bench/env/lib/python3.11/site-packages/click/decorators.py", line 21, in new_func
    return f(get_current_context(), *args, **kwargs)
      args = ()
      kwargs = {'skip_failing': False, 'skip_search_index': False}
      f = <function migrate at 0x7fd2227f4680>
  File "/home/frappe/frappe-bench/apps/frappe/frappe/commands/__init__.py", line 29, in _func
    ret = f(frappe._dict(ctx.obj), *args, **kwargs)
      ctx = <click.core.Context object at 0x7fd2227dfad0>
      args = ()
      kwargs = {'skip_failing': False, 'skip_search_index': False}
      profile = False
      f = <function migrate at 0x7fd2227f45e0>
  File "/home/frappe/frappe-bench/apps/frappe/frappe/commands/site.py", line 560, in migrate
    ).run(site=site)
      context = {'sites': ['localhost'], 'force': False, 'verbose': False, 'profile': False}
      skip_failing = False
      skip_search_index = False
      activate_by_import = <module 'traceback_with_variables.activate_by_import' from '/home/frappe/frappe-bench/env/lib/python3.11/site-packages/traceback_with_variables/activate_by_import.py'>
      SiteMigration = <class 'frappe.migrate.SiteMigration'>
      site = 'localhost'
  File "/home/frappe/frappe-bench/apps/frappe/frappe/migrate.py", line 166, in run
    frappe.init(site=site)
      self = <frappe.migrate.SiteMigration object at 0x7fd22179b790>
      site = 'localhost'
  File "/home/frappe/frappe-bench/apps/frappe/frappe/__init__.py", line 248, in init
    setup_module_map()
      site = 'localhost'
      sites_path = '.'
      new_site = False
  File "/home/frappe/frappe-bench/apps/frappe/frappe/__init__.py", line 1538, in setup_module_map
    for module in get_module_list(app):
      _cache = RedisWrapper<ConnectionPool<Connection<host=10.20.1.11,port=6379,db=0>>>
      app = 'erpnext'
      module = 'event_streaming'
  File "/home/frappe/frappe-bench/apps/frappe/frappe/__init__.py", line 1378, in get_module_list
    return get_file_items(os.path.join(os.path.dirname(get_module(app_name).__file__), "modules.txt"))
      app_name = 'erpnext'
  File "/home/frappe/frappe-bench/apps/frappe/frappe/__init__.py", line 1327, in get_module
    return importlib.import_module(modulename)
      modulename = 'erpnext'
  File "/usr/local/lib/python3.11/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
      name = 'erpnext'
      package = None
      level = 0
  File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
      name = 'erpnext'
      package = None
      level = 0
  File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
      name = 'erpnext'
      import_ = <function _gcd_import at 0x7fd2252a3d80>
      module = <object object at 0x7fd2252d8050>
  File "<frozen importlib._bootstrap>", line 1140, in _find_and_load_unlocked
      name = 'erpnext'
      import_ = <function _gcd_import at 0x7fd2252a3d80>
      path = None
      parent = ''
      parent_spec = None
      spec = None
builtins.ModuleNotFoundError: No module named 'erpnext'

 
```

**Comment by revant** [comment]
https://github.com/frappe/helm/actions/runs/5419865585/job/14677512184

These tests also test bench migrate

https://github.com/frappe/helm/blob/4986de9a91441b042b0454c1aff46e9469f26771/tests/script.sh#L71

**Comment by dungtc** [comment]
Thank @revant I just update the stack trace when start the worker, it doesn't find the erpnext module on frappe init_site, do you know why?

**Comment by dungtc** [comment]
Dear @revant  , please help. Thanks

**Comment by revant** [comment]
Give me access to apps and container registry to push image

**Comment by dungtc** [comment]
> Give me access to apps and container registry to push image

I use your docker image in here https://github.com/frappe/frappe_docker/blob/main/images/production/Containerfile
and run with erpnext helm chart.

My current version is v14.30.0 that includes Frappe, ERPNext.
When the run the deployment of worker or migration job I got this error `ModuleNotFoundError: No module named 'erpnext'` with this stacktrace.

I've checked my docker image, it already have erpnext, payments, frappe in apps folder and still don't know thy.
<img width="408" alt="image" src="https://github.com/frappe/frappe_docker/assets/52595737/b3709683-35eb-4b45-a36a-dc97abb5343c">
